### ISSUE TITLE
Module not found: Error: Can't resolve '~/package.json' in ...

### ISSUE THREAD

**Issue by paolodina** [tralves/nativescript-vue-webpack-template#17]
Hello, this is the output of `webpack --watch --env.tns --env.android`.

I can't say if it's really an issue, because the application starts and run on the emulator without any problem. IIRC I got the same messages even before your last commits.

```
paolo@tk:hello-ns-vue $ tns create hello-ns-vue --template https://github.com/tralves/nativescript-vue-webpack-template
paolo@tk:tmp $ cd hello-ns-vue/
paolo@tk:hello-ns-vue $ webpack --watch --env.tns --env.android

Webpack is watching the files…

(node:22188) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
Hash: 412e5b51b04e54721432
Version: webpack 2.6.1
Time: 5099ms
           Asset       Size  Chunks                    Chunk Names
       bundle.js    11.2 kB       0  [emitted]         bundle
      app.css.js  267 bytes       1  [emitted]         app.css
       vendor.js    1.53 MB       2  [emitted]  [big]  vendor
         app.css    13.9 kB          [emitted]         
images/apple.jpg     277 kB          [emitted]  [big]  
  images/vue.png    6.62 kB          [emitted]         
    package.json   89 bytes          [emitted]         
      starter.js   41 bytes          [emitted]         
  [22] ../~/tns-core-modules/ui/layouts/stack-layout/stack-layout.android.js 1.01 kB {2} [built]
  [23] ../~/tns-core-modules/ui/placeholder/placeholder.android.js 698 bytes {2} [built]
  [26] ../~/tns-core-modules/http/http.js 1.83 kB {2} [built]
  [27] ../~/nativescript-vue/dist/index.js 258 kB {2} [built]
 [116] ./JsComponent.js 86 bytes {0} [built]
 [117] ./vendor-platform.android.js 74 bytes {2} [built]
 [118] ./app.scss 41 bytes {0} [built]
 [119] ./VueComponent.vue 2.12 kB {0} [built]
 [120] ../~/tns-core-modules/bundle-entry-points.js 3.29 kB {2} [built]
 [121] ./app.js 515 bytes {0} [built]
 [122] ./vendor.js 102 bytes {2} [built]
 [124] ./app.css 41 bytes {1} [built]
 [152] ../~/tns-core-modules/ui/layouts/wrap-layout/wrap-layout.android.js 1.41 kB {2} [built]
 [164] ../~/nativescript-css-loader?{"minimize":false}!../~/nativescript-theme-core/css/core.light.android.css 13.9 kB [built]
 [165] ../~/nativescript-css-loader/lib/css-base.js 1.51 kB [built]
    + 153 hidden modules

WARNING in ../~/tns-core-modules/profiling/profiling.js
Module not found: Error: Can't resolve '~/package.json' in '/tmp/hello-ns-vue/node_modules/tns-core-modules/profiling'
 @ ../~/tns-core-modules/profiling/profiling.js 113:20-45
 @ ../~/tns-core-modules/ui/frame/frame.android.js
 @ ../~/tns-core-modules/ui/frame/activity.android.js
 @ ./vendor-platform.android.js
 @ ./vendor.js

WARNING in ../~/tns-core-modules/ui/styling/style-scope.js
Module not found: Error: Can't resolve '~/package.json' in '/tmp/hello-ns-vue/node_modules/tns-core-modules/ui/styling'
 @ ../~/tns-core-modules/ui/styling/style-scope.js 24:20-45
 @ ../~/tns-core-modules/ui/page/page-common.js
 @ ../~/tns-core-modules/ui/page/page.android.js
 @ ../~/tns-core-modules/bundle-entry-points.js
 @ ./vendor.js
Child extract-text-webpack-plugin:
       [0] ../~/nativescript-css-loader/lib/css-base.js 1.51 kB {0} [built]
       [1] ../~/nativescript-css-loader?{"minimize":false}!../~/nativescript-theme-core/css/core.light.android.css 13.9 kB {0} [built]
       [2] ../~/resolve-url-loader?{"silent":true,"absolute":false,"sourceMap":false,"fail":false,"keepQuery":false,"debug":false,"root":null}!../~/nativescript-css-loader?{"minimize":false}!../~/nativescript-dev-webpack/platform-css-loader.js!./app.css 390 bytes {0} [built]
Child extract-text-webpack-plugin:
       [0] ../~/css-loader/lib/css-base.js 2.26 kB {0} [built]
       [1] ../~/css-loader?{"url":false}!../~/sass-loader/lib/loader.js!./app.scss 190 bytes {0} [built]
Child extract-text-webpack-plugin:
       [0] ../~/css-loader/lib/css-base.js 2.26 kB {0} [built]
       [1] ../~/css-loader?{"url":false}!../~/ns-vue-loader/lib/style-compiler?{"vue":true,"id":"data-v-0f0e8576","scoped":true,"hasInlineConfig":false}!../~/sass-loader/lib/loader.js!../~/ns-vue-loader/lib/selector.js?type=styles&index=0&bustCache!./VueComponent.vue 209 bytes {0} [built]
Hash: 90c28bf983fc71c67f9d
Version: webpack 2.6.1
Time: 111ms
       Asset      Size  Chunks             Chunk Names
     app.css   13.9 kB          [emitted]  
package.json  89 bytes          [emitted]  
  starter.js  41 bytes          [emitted]  
    + 161 hidden modules

WARNING in ../~/tns-core-modules/profiling/profiling.js
Module not found: Error: Can't resolve '~/package.json' in '/tmp/hello-ns-vue/node_modules/tns-core-modules/profiling'
 @ ../~/tns-core-modules/profiling/profiling.js 113:20-45
 @ ../~/tns-core-modules/ui/frame/frame.android.js
 @ ../~/tns-core-modules/ui/frame/activity.android.js
 @ ./vendor-platform.android.js
 @ ./vendor.js

WARNING in ../~/tns-core-modules/ui/styling/style-scope.js
Module not found: Error: Can't resolve '~/package.json' in '/tmp/hello-ns-vue/node_modules/tns-core-modules/ui/styling'
 @ ../~/tns-core-modules/ui/styling/style-scope.js 24:20-45
 @ ../~/tns-core-modules/ui/page/page-common.js
 @ ../~/tns-core-modules/ui/page/page.android.js
 @ ../~/tns-core-modules/bundle-entry-points.js
 @ ./vendor.js
```

**Comment by tralves** [comment]
Yeah, those warnings are bothering me too. Didn't have time to look into that. Let's leave this issue here until someone figures it out.

**Comment by paolodina** [comment]
Maybe useful for more detail:
```
$ webpack --watch --env.tns --env.android --display-error-details
```

**Comment by paolodina** [comment]
News. I noted that it's like if the tilde symbol doesn't get expanded or it is in the wrong way.

> Module not found: Error: Can't resolve '~/package.json' in '/tmp/hello-ns-vue/node_modules/tns-core-modules/ui/styling'

In fact, if you add `.` to the modules list in `webpack.config.js` and do this:
```
$ pwd
/tmp/hello-ns-vue
mkdir ./~
$ cp package.json ./~/
```
the `webpack --watch --env.tns --env.android` command compiles just fine. Obviously we can't create the `./~` directory as a part of the project and copy/symlink package.json there, but I see there is an [`alias`](https://webpack.js.org/configuration/resolve/#resolve-alias) directive and wonder if we could use something like:
```
...
resolve: {
  alias: {
    '~': '.'
  },
  ...
},
...
```
It works, but to me it seems more a workaround than a solution.  What do you think?

**Comment by tralves** [comment]
hum... maybe that's may the the right actually. 
Maybe `'~': path.join(__dirname, 'tns')` would be better.

**Comment by tralves** [comment]
Fixed with #18.
